client/store/helpers/resource.js

```diff
@@ -44,8 +44,10 @@ export default class Resource {
    */
   processEntries(items) {
     return reduce(items, (acc, it) => {
-      this.setCid(it);
-      acc[it._cid] = it;
+      if (!this.getCid(it.id)) {
```

Comment:
This is being called only after fetch, when we know that cid needs to be set. `setCid` will attach mapping if it exists. Am I missing something / Why if ?

Reply:
For this one:


```vue
// helpers/resource.js
  fetch(params) {
    return this.get('', params)
      .then(({ data: { data } }) => this.processEntries(data));
  }

get(path, params = {}) {
    return axios.get(this.url(path), { params });
  }
```
This part fetches all the comments. It doesn't care if there is something in the state already. 
So it loads comment1,comment2,comment3 even if the comment1 was added in the store and state by the SSE. Then we have the duplication of the comment1. We could also remove this.setCid(it) from the if. But what is the purpose of setting a _cid that is already mapped?